ggshield/scan/repo.py

```diff
@@ -75,13 +77,56 @@ def scan_commit(
     except Exception as exc:
         results = Results.from_exception(exc)
 
-    return ScanCollection(
-        commit.sha or "unknown",
-        type="commit",
-        results=results,
-        optional_header=commit.optional_header,
-        extra_info=commit.info._asdict(),
-    )
+    scans = []
+    for commit in commits:
+        concerned_results = [
+            res
+            for res in results.results
+            if any(res.content == file.document for file in commit.files)
+        ]
+        scans.append(
+            ScanCollection(
+                commit.sha or "unknown",
+                type="commit",
+                results=Results(
+                    results=concerned_results,
+                    errors=results.errors,
+                ),
+                optional_header=commit.optional_header,
+                extra_info=commit.info._asdict(),
+            )
+        )
+
+    return ScanCollection(id=scan_context.command_id, type="commit-ranges", scans=scans)
+
+
+def get_commits_content_by_batch(
+    commit_list: List[str],
+    exclusion_regexes: Set[re.Pattern],
+    batch_max_size: int = MULTI_DOCUMENT_LIMIT,
+) -> Iterator[Tuple[Files, List[Commit]]]:
+    """
+    Given a list of commit shas yield the commit files
+    by biggest batches possible of length at most MULTI_DOCUMENT_LIMIT
+    """
+    current_files_batch: List[File] = []
+    current_commits_batch: List[Commit] = []
+    call_counter = 0
```

Comment:
I think this can be removed

Reply:
Yep, this was not exactly ready yet 😅   
I added a test and cleaned this. Should be better.

Reply:
Oups... 🚀